site-scrapers/StopAndShop.js

```diff
@@ -0,0 +1,71 @@
+const sites = require("../data/sites.json");
+
+const noAppointmentMatchString = "no locations with available appointments";
+
+module.exports = async function GetAvailableAppointments(browser) {
+    console.log("StopAndShop starting.");
+    const webData = await ScrapeWebsiteData(browser);
+    console.log("StopAndShop done.");
+    return sites.StopAndShop.locations.map((loc) => {
+        const response = webData[loc.zip];
+        return {
+            name: `Stop & Shop (${loc.city})`,
+            hasAvailability: response.indexOf(noAppointmentMatchString) == -1,
+            extraData: response.length
+                ? response.substring(1, response.length - 1)
+                : response, //take out extra quotes
+            signUpLink: sites.StopAndShop.website,
+            ...loc,
+        };
+    });
+};
+
+async function ScrapeWebsiteData(browser) {
+    const page = await browser.newPage();
+    await page.goto(sites.StopAndShop.website);
+    await page.solveRecaptchas().then(({ solved }) => {
+        if (solved.length) {
+            return page.waitForNavigation();
+        } else {
+            return;
+        }
+    });
+
+    const results = {};
+
+    for (const loc of [...new Set(sites.StopAndShop.locations)]) {
+        if (!results[loc.zip]) {
+            await page.evaluate(
+                () => (document.getElementById("zip-input").value = "")
+            );
+            await page.type("#zip-input", loc.zip);
+            const [searchResponse, ...rest] = await Promise.all([
+                Promise.race([
+                    page.waitForResponse(
+                        // TODO - this link doesn't work for me... fix? 
+                        "https://stopandshopsched.rxtouch.com/rbssched/program/covid19/Patient/CheckZipCode"
```

Comment:
This site doesn't work after the first time I visit it in my browser... wonder if that's related to my issue. But I just followed what we do for Hannaford, which doesn't generate the error.

Reply:
This is the underlying API call made from the site - so it doesn't matter if it doesn't work when I visit the url directly in my browser!